hydrosdk/predictor.py

```diff
@@ -1,72 +1,165 @@
-# TODO: re-write predictors
+from abc import abstractmethod, ABC
+from typing import Union, Dict
 
-import abc
-import requests
-import hydro_serving_grpc as hsg
-import hydro_serving_grpc.gateway as hsgateway
+import numpy as np
+import pandas as pd
+from hydro_serving_grpc import PredictionServiceStub, ModelSpec, predict_pb2, PredictResponse
+from hydro_serving_grpc.contract import ModelSignature
+from hydro_serving_grpc.gateway import GatewayServiceStub
 
+from hydrosdk.data.conversions import convert_inputs_to_tensor_proto
+from hydrosdk.data.types import PredictorDT, proto2np_dtype, DTYPE_TO_FIELDNAME
 
-class AbstractPredictor(abc.ABC):
-    @abc.abstractmethod
-    def predict(self, data):
+
+class PredictImplementation(ABC):
+    @abstractmethod
+    def send_data(self, model_spec, inputs):
         pass
 
+    def data_to_predictRequest(self, model_spec, inputs) -> predict_pb2.PredictRequest:
+        return predict_pb2.PredictRequest(model_spec=model_spec, inputs=inputs)
+
+
+class MetricableImplementation(PredictImplementation):
```

Comment:
🔍  Just a nitpicking, but I don't like the choice of words here 🤔 
It's not related to metric per se. Here we want to differentiate between data requests which could be monitored and not.
Perhaps, `MonitorableImplementation` and `UnmonitorableImplementation` (`RawImplementation`) ?

Reply:
Done